src/frontend/components/StudentLiveControlButtons/StudentJoinDiscussionButton/index.spec.tsx

```diff
@@ -0,0 +1,191 @@
+    fireEvent.change(inputUsername, { target: { value: 'Joe' } });
+    // then the confirm button is enabled
+    expect(confirmButton).toBeEnabled();
+
+    // clicking on cancel button remove the input and the ask button is displayed again
+    fireEvent.click(cancelButton);
+    await screen.findByRole('textbox', { name: 'username' });
+    expect(inputUsername).not.toBeInTheDocument();
+    screen.getByRole('button', {
+      name: 'Send request to join the discussion',
+    });
+    screen.getByText('join discussion icon');
+  });
+
+  it('displays the username input text and clicks on the confirm button', async () => {
+    render(wrapInIntlProvider(<StudentJoinDiscussionButton />));
+    mockAskParticipantToJoin
+      .mockRejectedValueOnce(Error('must be in the room before asking to join'))
+      .mockResolvedValueOnce();
+
+    const askButton = screen.getByRole('button', {
+      name: 'Send request to join the discussion',
+    });
+    screen.getByText('join discussion icon');
+    expect(
+      screen.queryByRole('textbox', { name: 'username' }),
+    ).not.toBeInTheDocument();
+
+    expect(useParticipantWorkflow.getState().asked).toEqual(false);
+    fireEvent.click(askButton);
+
+    expect(mockAskParticipantToJoin).toHaveBeenCalled();
+
+    const inputUsername = await screen.findByRole('textbox', {
+      name: 'username',
+    });
+    const confirmButton = screen.getByRole('button', { name: /confirm/i });
+    screen.getByRole('button', { name: /cancel/i });
+
+    // at first rendering the confirm button is disabled
+    expect(confirmButton).toBeDisabled();
+
+    // the input value is filled
+    fireEvent.change(inputUsername, { target: { value: 'Doe' } });
+    // then the confirm button is enabled
+    expect(confirmButton).toBeEnabled();
+
+    fireEvent.click(confirmButton);
+
+    await waitFor(() =>
+      expect(useParticipantWorkflow.getState().asked).toEqual(true),
+    );
+  });
+
+  it('renders the form with proper style without userName', async () => {
+    render(wrapInIntlProvider(<StudentJoinDiscussionButton />));
+    mockAskParticipantToJoin.mockRejectedValueOnce(
+      Error('must be in the room before asking to join'),
+    );
+
+    fireEvent.click(
+      screen.getByRole('button', {
+        name: 'Send request to join the discussion',
+      }),
+    );
+    await screen.findByRole('textbox', {
+      name: 'username',
+    });
+  });
+
+  it('renders the form with proper style with userName', async () => {
+    render(wrapInIntlProvider(<StudentJoinDiscussionButton />));
+    mockAskParticipantToJoin.mockRejectedValueOnce(
+      Error('must be in the room before asking to join'),
+    );
+
+    fireEvent.click(
+      screen.getByRole('button', {
+        name: 'Send request to join the discussion',
+      }),
+    );
+    const inputUsername = await screen.findByRole('textbox', {
+      name: 'username',
+    });
+    fireEvent.change(inputUsername, { target: { value: 'Joe' } });
```

Comment:
It seems like an assertion is missing after this event.

Reply:
yep, this test (and the test above) where there juste to snapshot the form, since the snapshot is removed (buggy) there is no purpose to keep those tests

Reply:
https://github.com/openfun/marsha/pull/1249/commits/ffdebc54aba425a9465a8be40a24263b84845556